[kube-prometheus] Allow to use go template for additionalPrometheusRules and alertmanager.templateFiles

### Name and Version

bitnami/kube-prometheus >= 7.0.0

### What steps will reproduce the bug?

1. Create custom-values.yaml

`custom-values.yaml`
```yaml
prometheus:
  additionalPrometheusRules:
    - name: custom-alerting-rules
      groups:
        - name: prometheus-config
          rules:
            - alert: PrometheusConfigurationReload
              expr: prometheus_config_last_reload_successful > 0
              for: 1m
              labels:
                severity: error
              annotations:
                summary: "Prometheus configuration reload (instance {{ $labels.instance }})"
                description: "Prometheus configuration reload error\n  VALUE = {{ $value }}\n  LABELS: {{ $labels }}"
```

Ref: https://github.com/bitnami/charts/blob/master/bitnami/kube-prometheus/values.yaml#L1035

2. Lint Chart

```bash
helm lint charts/bitnami/kube-prometheus -f custom-values.yaml
```

3. Temporary solution

Escaping prometheus rules

```yaml
prometheus:
  ## @param prometheus.additionalPrometheusRules PrometheusRule defines recording and alerting rules for a Prometheus instance.
  additionalPrometheusRules:
    - name: custom-alerting-rules
      groups:
        - name: prometheus-config
          rules:
            - alert: PrometheusConfigurationReload
              expr: prometheus_config_last_reload_successful > 0
              for: 1m
              labels:
                severity: error
              annotations:
                summary: "Prometheus configuration reload (instance {{`{{ $labels.instance }}`}})"
                description: "Prometheus configuration reload error\n  VALUE = {{`{{ $value }}`}}\n  LABELS: {{`{{ $labels }}`}}"
```

4. Solution

* Revert `templates/alertmanager/secrets.yaml` and `templates/prometheus/additionalPrometheusRules.yaml`
* Add new Values: `prometheus.additionalRawPrometheusRules` and `alertmanager.rawFiles`


### Are you using any custom parameters or values?

```yaml
prometheus:
  ## @param prometheus.additionalPrometheusRules PrometheusRule defines recording and alerting rules for a Prometheus instance.
  additionalPrometheusRules:
    - name: custom-alerting-rules
      groups:
        - name: prometheus-config
          rules:
            - alert: PrometheusConfigurationReload
              expr: prometheus_config_last_reload_successful > 0
              for: 1m
              labels:
                severity: error
              annotations:
                summary: "Prometheus configuration reload (instance {{ $labels.instance }})"
                description: "Prometheus configuration reload error\n  VALUE = {{ $value }}\n  LABELS: {{ $labels }}"
```

### What is the expected behavior?

Prometheus rules and AlertManager configurations contain many go templates. It's strange not to put them raw anymore.

It causes a lot of problems. Potentially, you would have to escape all the lines

### What do you see instead?

```console
==> Linting charts/bitnami/kube-prometheus/

[ERROR] templates/: template: kube-prometheus/templates/prometheus/additionalPrometheusRules.yaml:17:14: executing "kube-prometheus/templates/prometheus/additionalPrometheusRules.yaml" at <include "common.tplvalues.render" (dict "value" .groups "context" $)>: error calling include: template: kube-prometheus/charts/common/templates/_tplvalues.tpl:11:12: executing "common.tplvalues.render" at <tpl (.value | toYaml) .context>: error calling tpl: error during tpl function execution for "- name: prometheus-config\n  rules:\n  - alert: PrometheusConfigurationReload\n    annotations:\n      description: |-\n        Prometheus configuration reload error\n          VALUE = {{ $value }}\n          LABELS: {{ $labels }}\n      summary: Prometheus configuration reload (instance {{ $labels.instance }})\n    expr: prometheus_config_last_reload_successful > 0\n    for: 1m\n    labels:\n      severity: error": parse error at (kube-prometheus/templates/prometheus/additionalPrometheusRules.yaml:7): undefined variable "$value"

Error: 1 chart(s) linted, 1 chart(s) failed
```


### Additional information

New `templates/alertmanager/secrets.yaml`

```yaml
{{- if (and .Values.alertmanager.enabled (not .Values.alertmanager.externalConfig) ) }}
apiVersion: v1
kind: Secret
metadata:
  name: alertmanager-{{ template "kube-prometheus.alertmanager.fullname" . }}
  namespace: {{ include "common.names.namespace" . | quote }}
  labels: {{- include "kube-prometheus.alertmanager.labels" . | nindent 4 }}
    {{- if .Values.commonLabels }}
    {{- include "common.tplvalues.render" (dict "value" .Values.commonLabels "context" $) | nindent 10 }}
    {{- end }}
  {{- if .Values.commonAnnotations }}
  annotations: {{- include "common.tplvalues.render" (dict "value" .Values.commonAnnotations "context" $) | nindent 10 }}
  {{- end }}
data:
  alertmanager.yaml: {{ toYaml .Values.alertmanager.config | b64enc | quote }}
  {{- range $key, $val := .Values.alertmanager.templateFiles }}
  {{ $key }}: {{ $val | b64enc | quote }}
  {{- end }}
{{- end }}
```

New `templates/prometheus/additionalPrometheusRules.yaml`

```yaml
{{- if and .Values.prometheus.enabled .Values.prometheus.additionalPrometheusRules }}
{{- range .Values.prometheus.additionalPrometheusRules }}
---
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  name: {{ template "kube-prometheus.name" $ }}-{{ .name }}
  namespace: {{ include "common.names.namespace" $ | quote }}
  labels: {{ include "kube-prometheus.prometheus.labels" $ | nindent 4 }}
    {{- if $.Values.commonLabels }}
    {{- include "common.tplvalues.render" ( dict "value" $.Values.commonLabels "context" $ ) | nindent 4 }}
    {{- end }}
  {{- if $.Values.commonAnnotations }}
  annotations: {{- include "common.tplvalues.render" ( dict "value" $.Values.commonAnnotations "context" $ ) | nindent 4 }}
  {{- end }}
spec:
  groups: {{- toYaml .groups | nindent 4 }}
{{- end }}
{{- end }}
```

--- a/bitnami/kube-prometheus/Chart.yaml
+++ b/bitnami/kube-prometheus/Chart.yaml
@@ -35,4 +35,4 @@ sources:
   - https://github.com/bitnami/bitnami-docker-prometheus
   - https://github.com/bitnami/bitnami-docker-alertmanager
   - https://github.com/prometheus-operator/kube-prometheus
-version: 8.0.1
+version: 8.0.2

--- a/bitnami/kube-prometheus/templates/alertmanager/secrets.yaml
+++ b/bitnami/kube-prometheus/templates/alertmanager/secrets.yaml
@@ -12,8 +12,8 @@ metadata:
   annotations: {{- include "common.tplvalues.render" (dict "value" .Values.commonAnnotations "context" $) | nindent 10 }}
   {{- end }}
 data:
-  alertmanager.yaml: {{ include "common.tplvalues.render" (dict "value" .Values.alertmanager.config  "context" $) | b64enc | quote }}
+  alertmanager.yaml: {{ toYaml .Values.alertmanager.config | b64enc | quote }}
   {{- range $key, $val := .Values.alertmanager.templateFiles }}
-  {{ $key }}: {{ include "common.tplvalues.render" (dict "value" $val "context" $) | b64enc | quote }}
+  {{ $key }}: {{ $val | b64enc | quote }}
   {{- end }}
 {{- end }}

--- a/bitnami/kube-prometheus/templates/prometheus/additionalPrometheusRules.yaml
+++ b/bitnami/kube-prometheus/templates/prometheus/additionalPrometheusRules.yaml
@@ -14,6 +14,6 @@ metadata:
   annotations: {{- include "common.tplvalues.render" ( dict "value" $.Values.commonAnnotations "context" $ ) | nindent 4 }}
   {{- end }}
 spec:
-  groups: {{- include "common.tplvalues.render" ( dict "value" .groups "context" $ ) | nindent 4 }}
+  groups: {{- toYaml .groups | nindent 4 }}
 {{- end }}
 {{- end }}

